src/swagger-generation/output/metadata.json

```diff
@@ -246,5 +246,143 @@
         ]
       }
     }
+  },
+  "0.1.9-preview": {
+    "users": {
+      "beta": {
+        "isIdempotent": false,
+        "isReadonly": true,
+        "updatable": false,
+        "isContainment": false
```

Comment:
metadata for users should indicate that `userPrincipalName` is the alternate key. It's probably that msgraph-metadata submodule is not the latest version therefore doesn't include the annotation. May need to update the submodule and regenerated the types and schema metadata

Reply:
@jason-dou Regenerated off latest Graph meta data - please recheck.

I think we need to add a note to the generation instructions that in the PR, the msgraph-metadata sub-module needs to be rebased from the latest master update.  What do you think?

Reply:
Yeah we should add that. Also I realized that I had a script to automate the process a bit https://github.com/microsoftgraph/msgraph-bicep-types/blob/main/scripts/UpdateGeneratedTypes.ps1. I will check if this still works with node v22 and if more automation steps can be added

Reply:
Yeah - that looks like it would help.  It needs updating to put the first `--` in the start and generate cmd lines in quotes to work with node v22.  LMK what other changes need to be made to the PS script and I can check those in together with a revised set of instructions.

Please check the rest of the PR to make sure it looks good.

Reply:
Yeah I think that's the only change needed in the script. The rest of the PR looks good except for the property flags for the `userPrincipalName` - I will make the change in a separate PR

Reply:
Updates made to simplify the instructions to use the PS script.
LMK if this is now good by approving and I can merge.